### PR TITLE
Link envelope to contact side modal

### LINKED ISSUE #237: Link listing forms to side modal form
From a listing card, when the envelope is clicked it should open the side modal form.
On a listing page, the contact, make an offer, and see the property all open the side modal form.

In most cases, but not all, information about the listing are displayed in the form.

Most of the time, the listing shows the office. Sometimes the agent is tied to the listing.
Screenshot 2024-05-23 at 1 09 54 PM
Screenshot 2024-05-23 at 1 10 37 PM

### PR DESCRIPTION
**Pull request by rrusher**
Added side modal link to property cards.
When an agent is specified, dynamically update the contact info. i.e. Featured or Luxury properties.
Refactored cards to dom-helpers.
Added styling to side-modal.

Fix #237 

Test URLs:
- Before: https://main--hsf-commonmoves--hlxsites.hlx.live/
- After: https://237-modal-link--hsf-commonmoves--hlxsites.hlx.live/

### CODE REVIEW

File: blocks/property-search-results/property-search-results.js [hunk 1]
```diff
@@ -103,6 +103,7 @@ async function doSearch(search, redraw = true) {
     const controller = searchController;
     propertySearch(search).then((results) => {
       if (!controller.signal.aborted) {
+        window.propertyListings = results;
```
Comment (bstopp) [inline blocks/property-search-results/property-search-results.js:106]: Why are we storing this?

File: blocks/property-listing/property-listing.js [hunk 2]
```diff
@@ -2,38 +2,43 @@ import { getMetadata, readBlockConfig } from '../../scripts/aem.js';
...
   }
 
-  const search = await Search.fromBlockConfig(config);
-  search.franchiseeCode = getMetadata('office-id');
-  const list = document.createElement('div');
-  list.classList.add('property-list-cards', `rows-${Math.floor(search.pageSize / 8)}`);
+  const list = div({ class: `property-list-cards rows-${Math.floor(search.pageSize / 8)}` });
   block.append(list);
   propertySearch(search).then((results) => {
+    window.propertyListings = results;
```
Comment (bstopp) [inline blocks/property-listing/property-listing.js:41]: Why are we storing this?

File: blocks/property-search-results/property-search-results.js (hunk 1 above)
Comment (rrusher) [reply blocks/property-search-results/property-search-results.js:106]: So I don't have to go to the server to get basic information that we already have.

File: blocks/property-listing/property-listing.js (hunk 2 above)
Comment (rrusher) [reply blocks/property-listing/property-listing.js:41]: So I don't have to go to the server to get basic information that we already have.

**Pull request merged**